### DISCUSSION TITLE
What is wrong in this type guard?

### DISCUSSION THREAD

**Discussion by iBobik** [vuejs/language-tools#3452] [Q&A]
Why type guards for this type does not work?

<img width="858" alt="Snímek obrazovky 2023-07-26 v 19 05 59" src="https://github.com/vuejs/language-tools/assets/614232/c8052703-dc4a-428c-acd0-a5098a655350">

```
export type OfferDraft = {
    id: any;
    price: number;
    quantity: number;
    dueDate: any;
    product?: {
        id: any;
        title: string;
        imageUrl?: string | null | undefined;
    } | null | undefined;
    size?: {
        id: any;
        title: string;
    } | null | undefined;
} | {
  sizeId: string
  price?: number
  quantity?: number
}
```

All this did not worked:
```
<el-input v-if="'size' in offers[$index]" :value="offers[$index].size?.title" disabled />
<el-select v-if="'sizeId' in offers[$index]" v-model="offers[$index].sizeId">
'size' in offers[$index] ? offers[$index].size.title : ''
```

**Comment by so1ve** [comment]
https://www.typescriptlang.org/play?#code/C4TwDgpgBA8gZnCAnAIkghnYUC8UDeAvlAD4EBQUUAzgJYBeEAkgCYBcNwStAdgOaUoYbgGMIAfg48ArgFsARskEBHaeh7BaoSVBkKlhcuREB7HtWxwTJjvESoMWANoBdXFCf5BVOo1YcAcgDyQhcjKyQoAApTc2xaKF4oKxMASgoqKlo4aIDfZhYAxJ5k6ydaF3SvTJqoWOoTABsIADpGkz4olPKXFvzWVO8oQ0IgA

TypeScript Limitation

**Reply by iBobik** [reply]
How to deal with it?
